File: sentimental-cash/cash.py
from cs50 import get_float

# Prompt for a valid, non-negative amount
while True:
    try:
        change_owed = float(input("Change owed: "))
        if change_owed >= 0:
            break
    except ValueError:
        pass  # Ignore non-numeric input; will re-prompt

# Convert dollars to cents and round to avoid floating-point inaccuracies
cents = round(change_owed * 100)

# Initialize counter for coins
coins = 0

# Calculate the minimum number of coins needed
for coin in [25, 10, 5, 1]:
    coins += cents // coin
    cents %= coin

# Output the result
print(coins)

File: cash/cash.c
#include <stdio.h>

int main(void)
{
    int cents;

    // Prompt the user for the amount of change owed
    do
    {
        printf("Change owed: ");
        if (scanf("%d", &cents) != 1) // Validate input is an integer
        {
            // Clear invalid input from buffer if scanf fails
            while (getchar() != '\n')
                ;
            cents = -1; // Set cents to an invalid value to repeat the loop
        }
    } while (cents < 0); // Keep prompting until a valid input is received

    // Calculate minimum coins needed
    int coins = 0;

    // Use the largest coin denominations first
    coins += cents / 25; // Quarters
    cents %= 25;

    coins += cents / 10; // Dimes
    cents %= 10;

    coins += cents / 5; // Nickels
    cents %= 5;

    coins += cents; // Pennies (remaining cents)

    // Output the result
    printf("%d\n", coins);

    return 0;
}
